cabby/model/text/dual_encoder/dataset_rvs.py

```diff
@@ -0,0 +1,172 @@
+      lambda x: gutil.coords_from_list_coord(x)).tolist()
+
+    data['cellid'] = data.point.apply(
+      lambda x: gutil.cellid_from_point(x, s2level))
+
+    self.labels = data.cellid.apply(lambda x: cellid_to_label[x]).tolist()
+
+    data['neighbor_cells'] = data.cellid.apply(
+      lambda x: gutil.neighbor_cellid(x))
+
+    data['far_cells'] = data.cellid.apply(
+      lambda cellid: unique_cells_df[unique_cells_df['cellid']==cellid].far.iloc[0])
+
+    cellids_array = np.array(data.cellid.tolist())
+    neighbor_cells_array = np.array(data.neighbor_cells.tolist())
+    far_cells_array = np.array(data.far_cells.tolist())
+
+    self.cellids = util.binary_representation(cellids_array, dim = CELLID_DIM)
+    self.neighbor_cells =  util.binary_representation(
+      neighbor_cells_array, dim = CELLID_DIM)
+
+    self.far_cells =  util.binary_representation(
+      far_cells_array, dim = CELLID_DIM)
+
+    
+  def __getitem__(self, idx: int):
+    '''Supports indexing such that TextGeoDataset[i] can be used to get 
+    i-th sample. 
+    Arguments:
+      idx: The index for which a sample from the dataset will be returned.
+    Returns:
+      A single sample including text, the correct cellid, a neighbor cellid, 
+      a far cellid, a point of the cellid and the label of the cellid.
+    '''
+    text = {key: torch.tensor(val[idx])
+        for key, val in self.encodings.items()}
+    cellid = torch.tensor(self.cellids[idx])
+    neighbor_cells = torch.tensor(self.neighbor_cells[idx])
+    far_cells = torch.tensor(self.far_cells[idx])
+    point = torch.tensor(self.points[idx])
+    label = torch.tensor(self.labels[idx])
+    
+    sample = {'text': text, 'cellid': cellid, 'neighbor_cells': neighbor_cells, 
+      'far_cells': far_cells, 'point': point, 'label': label}
+
+    return sample
+
+  def __len__(self):
+    return len(self.cellids)
+
+
+def create_dataset(
+          data_dir: Text, 
+          region: Text, 
+          s2level: int, 
+          infer_only: bool = False
+) -> dataset_item.TextGeoDataset:
+  '''Loads data and creates datasets and train, validate and test sets.
+  Arguments:
+    data_dir: The directory of the data.
+    region: The region of the data.
+    s2level: The s2level of the cells.
+  Returns:
+    The train, validate and test sets and the dictionary of labels to cellids.
+  '''
+  ds = pd.read_json(os.path.join(data_dir, 'dataset.json'))
+  logging.info(f"Size of dataset before removal of duplication: {ds.shape[0]}")
+  ds = ds.drop_duplicates(subset=['end_osmid', 'start_osmid'], keep='last')
+  logging.info(f"Size of dataset after removal of duplication: {ds.shape[0]}")
+  dataset_size = ds.shape[0]
+  train_size = round(dataset_size*80/100)
+  valid_size = round(dataset_size*10/100)
+
+  train_ds = ds.iloc[:train_size]
+  valid_ds = ds.iloc[train_size:train_size+valid_size]
+  test_ds = ds.iloc[train_size+valid_size:]
+
+  # Get labels.
+  get_region = regions.get_region(region)
```

Comment:
`get_region` is not a good variable name here. Either `region` or something like `active_region` would work.